### ISSUE TITLE
Unable to open IPython console in Spyder

### ISSUE THREAD

**Issue by babykiss4ever** [spyder-ide/spyder#2444]
## Description

* *What steps will reproduce the problem?*

1. Install Elementary OS(the latest version)
2. Install Anaconda
3. Type `spyder` in terminal to open the IDE 

* *What is the expected output? What do you see instead?*
Expected: Open the IPython console
Error message in the panel:
```
'import sitecustomize' failed; use ‑v for traceback
Traceback (most recent call last):
File "/home/lumit/anaconda/lib/python2.7/site‑packages/spyderlib/widgets/externalshell/start_ipython_kernel.py", line 163, in 
ipk_temp.config = kernel_config()
File "/home/lumit/anaconda/lib/python2.7/site‑packages/spyderlib/widgets/externalshell/start_ipython_kernel.py", line 27, in kernel_config
from spyderlib.config import CONF
File "/home/lumit/anaconda/lib/python2.7/site‑packages/spyderlib/config.py", line 22, in 
from spyderlib.utils import iofuncs, codeanalysis
File "/home/lumit/anaconda/lib/python2.7/site‑packages/spyderlib/utils/iofuncs.py", line 26, in 
import pandas as pd
File "/home/lumit/anaconda/lib/python2.7/site‑packages/pandas/__init__.py", line 49, in 
from pandas.core.api import *
File "/home/lumit/anaconda/lib/python2.7/site‑packages/pandas/core/api.py", line 9, in 
from pandas.core.groupby import Grouper
File "/home/lumit/anaconda/lib/python2.7/site‑packages/pandas/core/groupby.py", line 15, in 
from pandas.core.frame import DataFrame
File "/home/lumit/anaconda/lib/python2.7/site‑packages/pandas/core/frame.py", line 39, in 
from pandas.core.series import Series
File "/home/lumit/anaconda/lib/python2.7/site‑packages/pandas/core/series.py", line 2702, in 
import pandas.tools.plotting as _gfx
File "/home/lumit/anaconda/lib/python2.7/site‑packages/pandas/tools/plotting.py", line 27, in 
import pandas.tseries.converter as conv
File "/home/lumit/anaconda/lib/python2.7/site‑packages/pandas/tseries/converter.py", line 8, in 
import matplotlib.dates as dates
File "/home/lumit/anaconda/lib/python2.7/site‑packages/matplotlib/dates.py", line 136, in 
import matplotlib.cbook as cbook
AttributeError: 'module' object has no attribute 'cbook'
```

* *Please provide any additional information below*

PS.
I can open the kernel by running `python start_ipython_kernel.py` and connect it with its id.

## Version and main components

* Spyder Version:  2.3.4
* Python Version:  2.7.9
* Qt Version    :  4.8.6, PyQt4 (API v2) 4.11.3 on Linux

## Optional dependencies

pyflakes >=0.5.0:  0.8.1 (OK)
pep8 >=0.6      :  1.6.2 (OK)
IPython >=1.0   :  3.1.0 (OK)
zmq >=2.1.11    :  14.5.0 (OK)
pygments >=1.6  :  2.0.2 (OK)
pandas >=0.13.1 :  0.15.2 (OK)
sphinx >=0.6.6  :  1.2.3 (OK)
psutil >=0.3    :  2.2.1 (OK)
rope >=0.9.2    :  0.9.4 (OK)
jedi >=0.8.1    :  0.8.1 (OK)
matplotlib >=1.0:  1.4.3 (OK)
sympy >=0.7.3   :  0.7.6 (OK)
pylint >=0.25   :  None (NOK)

**Comment by ccordoba12** [comment]
This seems an installation error with Anaconda. My suggestion would be to:

1. Remove your current installation.
2. Download Anaconda again.
3. Reinstall and try to run Spyder again.

**Comment by babykiss4ever** [comment]
@ccordoba12  

I tried to reinstall, yet it didn't work.
By googling the error message, this bug may result from 'circular import', but it's hard to troubleshoot the critical line in those files.

**Comment by ccordoba12** [comment]
Then try to run

    conda install -f matplotlib
    conda install -f pandas

and try again. maybe that fixes your problem :-)

**Comment by babykiss4ever** [comment]
@ccordoba12 

It didn't work :(. The error message shows that `sitecustomize.py` is not correct, but the top of the trace stack is `start_ipython_kernel.py`. 
However, by running it independently no error messages occur.

//-----------------------
I just fixed this bug by *commenting* line 217 - line 220:
```python
try:
    import matplotlib
except ImportError:
    matplotlib = None   # analysis:ignore
```

Yet it may cause issue #2257 according to the comments,
so maybe we can detect the version of python(my python version is 2.7 and #2257's is 3+) and even the OS version(each of two only occur in Linux) to decide whether we should `import matplotlib` here.

**Comment by ccordoba12** [comment]
The problem seems to be with Pandas, not with Matplotlib, as your traceback clearly shows.

I don't know why this is happening, but we protected Spyder against this problem in master. However, the frequency of this problem seems to justify releasing **2.3.5**

**Comment by ccordoba12** [comment]
This is a duplicate of issue #2456, and a fix for it will be released on 2.3.5
